cloud/aws/aws_s3.cc

```diff
@@ -477,7 +481,11 @@ Status S3StorageProvider::CreateBucket(const std::string& bucket) {
   Aws::S3::Model::BucketLocationConstraint bucket_location = Aws::S3::Model::
       BucketLocationConstraintMapper::GetBucketLocationConstraintForName(
           ToAwsString(env_->GetCloudEnvOptions().dest_bucket.GetRegion()));
-  if (bucket_location != Aws::S3::Model::BucketLocationConstraint::NOT_SET) {
+  // to create a bucket in US-EAST-1 you must specify no constraint location for
+  // legacy reasons
+  if ((bucket_location != Aws::S3::Model::BucketLocationConstraint::NOT_SET) &&
+      (bucket_location !=
+       Aws::S3::Model::BucketLocationConstraint::us_east_1)) {
```

Comment:
curious to know why we need to set us_east_1 specially

Reply:
It's a problem with the US EAST 1 API. We've encountered because we run mostly on US EAST 1.

Reply:
https://docs.aws.amazon.com/AmazonS3/latest/API/API_CreateBucket.html

_By default, the bucket is created in the US East (N. Virginia) Region. You can optionally specify a Region in the request body. You might choose a Region to optimize latency, minimize costs, or address regulatory requirements._

The phrasing is not clear, but in fact, what it means is that if you create a bucket in US EAST 1, you don't specify a region. You specify a region only if you don't create it in US EAST 1.

Reply:
Got it, thanks. If you can put the aws docs link as code comments as well, that will be great.

Reply:
Great idea!